### PR TITLE
Add snyk scan periodic CI job

### PR DESCRIPTION
**Pull request by navidshaikh**
- This changeset adds a periodic CI job to run snyk scan against
   kubernetes master every 6 hours. Also adds 'sig-security-snyk-scan'
   dashboard to testgrid.
 - It installs snyk CLI and requires 'snyk_token' secret available
   in test infra with key name 'SNYK_TOKEN'. This secret is used to
   populate env var 'SNYK_TOKEN', required for snyk CLI auth.
 - Licenses and few false positive deps (eg version '0.0.0') are
   filtered from the snyk scan results and printed on stdout as well
   as "${ARTIFACTS}/snyk_results.json"

xref: https://github.com/kubernetes/kubernetes/issues/101528

### CODE REVIEW

File: config/jobs/kubernetes/sig-security/snyk-scan.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        - -c
+        - |
+          set -euo pipefail; \
+          apt update && apt -y install jq; \
+          wget -O /usr/local/bin/snyk https://github.com/snyk/snyk/releases/download/v1.605.0/snyk-linux && chmod +x /usr/local/bin/snyk
+          mkdir -p "${ARTIFACTS}"; \
+          snyk test --json  | jq '.vulnerabilities | .[] | select ((.type=="license") or (.version=="0.0.0") | not)' | tee "${ARTIFACTS}/snyk_results.json"
+    annotations:
+      testgrid-create-test-group: "true"
+      testgrid-dashboards: sig-security-snyk-scan
```
Comment (navidshaikh) [inline config/jobs/kubernetes/sig-security/snyk-scan.yaml:37]: what'd be the correct dashboard to use here ? ( the listed one doesn't exist)

Comment (dims) [reply config/jobs/kubernetes/sig-security/snyk-scan.yaml:37]: let's do this in `sig-testing-misc` for now.

Comment (dims) [reply config/jobs/kubernetes/sig-security/snyk-scan.yaml]: if you want a new one, follow the pattern here: https://cs.k8s.io/?q=sig-security-cvelist-public&i=nope&files=&excludeFiles=&repos=kubernetes/test-infra

File: config/jobs/kubernetes/sig-security/snyk-scan.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#  Periodic CI job for running snyk scans against k/k master
+# - It installs snyk CLI and requires 'snyk_token' secret available
+#   in test infra with key name 'SNYK_TOKEN'. This secret is used to
+#   populate env var 'SNYK_TOKEN', required for snyk CLI auth.
+# - Licenses and few false positive deps (eg version '0.0.0') are
+#   filtered from the snyk scan results and printed on stdout as well
+#   as "${ARTIFACTS}/snyk_results.json"
+periodics:
+  - interval: 6h
+    name: snyk-scan-kubernetes-master-periodical
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-security/snyk-scan.yaml:10]: Run in k8s-infra-prow-build trusted, and follow job naming convention of kubernetes periodics being named as `ci-kubernetes-{foo}-{branch}`
```suggestion
    cluster: k8s-infra-prow-build-trust
    name: ci-kubernetes-snyk-master
```

**Review by spiffxp — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes/sig-security/snyk-scan.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      repo: kubernetes
+      base_ref: master
+      path_alias: k8s.io/kubernetes
+    spec:
+      containers:
+      - image: golang
+        envFrom:
+        - secretRef:
+            # secret key should be defined as SNYK_TOKEN
+            name: snyk_token
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-security/snyk-scan.yaml:23]: https://github.com/kubernetes/k8s.io/pull/2222 adds this to `k8s-infra-prow-build-trusted`

**Comment by navidshaikh** [comment]
/retitle Add snyk scan periodic CI job

File: config/jobs/kubernetes/sig-security/snyk-scan.yaml (hunk 2 above)
Comment (navidshaikh) [reply config/jobs/kubernetes/sig-security/snyk-scan.yaml]: done

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by navidshaikh** [comment]
@spiffxp : moved to job spec to `config/jobs/kubernetes/wg-k8s-infra/trusted/wg-k8s-infra-trusted.yaml` as `TestK8sInfraTrusted` test assert

https://github.com/kubernetes/test-infra/blob/cf184f2204cea2bba5e26d0d44c5f986802fb2bd/config/tests/jobs/jobs_test.go#L349-L350

**Pull request merged**